src/main/java/com/github/steveice10/mc/protocol/data/game/inventory/CreativeGrabAction.java

```diff
@@ -1,5 +1,30 @@
 package com.github.steveice10.mc.protocol.data.game.inventory;
 
+import it.unimi.dsi.fastutil.ints.Int2ObjectMap;
+import it.unimi.dsi.fastutil.ints.Int2ObjectOpenHashMap;
+
 public enum CreativeGrabAction implements ContainerAction {
-    GRAB;
+    GRAB(2);
+
+    private final int id;
+
+    CreativeGrabAction(int id) {
+        this.id = id;
+    }
+
+    public int getId() {
+        return id;
+    }
+
+    private static Int2ObjectMap<CreativeGrabAction> VALUES = new Int2ObjectOpenHashMap<>();
```

Comment:
If we know there's only one item, let's simplify the logic and not allocate a whole map.